src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -1120,19 +1122,51 @@ void SpatialPooler::updateDutyCyclesHelper_(vector<Real>& dutyCycles,
 
 void SpatialPooler::updateBoostFactors_()
 {
-  for (UInt i = 0; i < numColumns_; i++)
+
+  if (maxBoost_ > 1.0)
   {
-    if (minActiveDutyCycles_[i] <= 0)
+    vector<Real> targetDensity(numColumns_, 0);
```

Comment:
This vector is really only needed in the local inhibition case. Maybe it'd be better to do the boostFactor computation in two places, and avoid creating this vector every time step in the "global inhibition" case?

In other words, there will be two different lines:

~~~c++
// Global inhibition
Real boostFactor = exp(-(activeDutyCycles_[i] - density) * maxBoost_);

// Local inhibition
Real boostFactor = exp(-(activeDutyCycles_[i] - targetDensity[i]) * maxBoost_);
 ~~~